crates/bevy_text/src/text2d.rs

```diff
@@ -95,47 +95,41 @@ pub fn extract_text2d_sprite(
         .map(|window| window.resolution.scale_factor() as f32)
         .unwrap_or(1.0);
 
-    for (entity, computed_visibility, text, text_layout_info, anchor, text_transform) in
+    for (entity, computed_visibility, text, text_layout_info, anchor, global_transform) in
         text2d_query.iter()
     {
         if !computed_visibility.is_visible() {
             continue;
         }
 
-        let text_glyphs = &text_layout_info.glyphs;
         let text_anchor = anchor.as_vec() * Vec2::new(1., -1.) - 0.5;
-        let alignment_offset = text_layout_info.size * text_anchor;
+        let alignment_translation = text_layout_info.size * text_anchor;
+        let transform = *global_transform
+            * Transform::from_scale(Vec3::splat(scale_factor.recip()))
+            * Transform::from_translation(alignment_translation.extend(0.));
```

Comment:
Should we be using `Transform` functions here instead of `GlobalTransform`? Seems like it has to compute the affine for this immediately after constructing it here.

It also seems like the scale transform here could be precomputed based on the scale factor outside of the loop.